src/client/ua_client_connect.c

```diff
@@ -1372,6 +1421,62 @@ UA_Client_connectSecureChannel(UA_Client *client, const char *endpointUrl) {
     return client->connectStatus;
 }
 
+#ifdef UA_ENABLE_ENCRYPTION
+/* Create a SecurityPolicy for authentication, which is used to create the userTokenSignature.
+ * A separate SecurityPolicy is required because the user can use a different certificate and
+ * private key than for the SecureChannel. The private key is stored in the SecurityPolicy. */
+static UA_StatusCode
+UA_Client_createPolicyAuth(UA_Client *client, UA_ByteString certificate, UA_ByteString privateKey) {
+    UA_ClientConfig *cc = UA_Client_getConfig(client);
+    cc->securityPolicyAuth = (UA_SecurityPolicy*) UA_malloc(sizeof(UA_SecurityPolicy));
+    UA_String aes128_Sha256_RsaOaep = UA_STRING("http://opcfoundation.org/UA/SecurityPolicy#Aes128_Sha256_RsaOaep");
+    UA_String basic128Rsa15  = UA_STRING("http://opcfoundation.org/UA/SecurityPolicy#Basic128Rsa15");
+    UA_String basic256  = UA_STRING("http://opcfoundation.org/UA/SecurityPolicy#Basic256");
+    UA_String basic256Sha256 = UA_STRING("http://opcfoundation.org/UA/SecurityPolicy#Basic256Sha256");
+
+    UA_StatusCode retval = UA_STATUSCODE_BAD;
+    if(UA_String_equal(&cc->securityPolicyUri, &aes128_Sha256_RsaOaep)) {
+        retval =
```

Comment:
Don't use any specific plugin in the core library.
The SecurityPolicies are already set up for the SecureChannels in the client config.
You can juse use the name to look up the correct policy.